src/web/stash/ItemTooltip.tsx

```diff
@@ -14,18 +16,52 @@ export function ItemTooltip({ item }: { item: Item }) {
     return <span class={className}>{item.name}</span>;
   }
 
+  const rangeMap = getRanges(item).reduce(function (
+    map: { [prop: string]: number[] },
```

Comment:
Typescript has a built-in type for this, `Record<string, number[]>`. But unless you need JSON serialization (like I do for the game data), I'd recommend [using `Map` instead](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map).

```
const rangeMap = getRanges(item)
  .reduce(function (map, range) { ... }, new Map<string, number[]>());
```